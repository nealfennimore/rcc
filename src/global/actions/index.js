import { createAction } from 'redux-actions';

export const EXAMPLE_ACTION = 'EXAMPLE_ACTION';
export const exampleAction = createAction( EXAMPLE_ACTION );
