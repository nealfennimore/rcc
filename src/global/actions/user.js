import { createAction } from 'redux-actions';

export const SET_USERNAME = 'SET_USERNAME';
export const setUsername = createAction( SET_USERNAME, ( username ) => ( { username } ) );

export const SET_TOKEN = 'SET_TOKEN';
export const setToken = createAction( SET_TOKEN, ( token ) => ( { token } ) );
