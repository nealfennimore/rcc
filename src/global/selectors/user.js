import { createSelector } from 'reselect';

const getSlice = ( state ) => state.user;

export const getUsername = createSelector( getSlice, ( user ) => user.username );
export const getToken = createSelector( getSlice, ( user ) => user.token );
