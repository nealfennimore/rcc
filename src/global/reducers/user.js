import { SET_USERNAME, SET_TOKEN } from 'global/actions/user';

const INITIAL_STATE = {
    username: null,
    token: null,
};

export default function user( state = INITIAL_STATE, action ) {
    switch ( action.type ) {
    case SET_USERNAME:
    case SET_TOKEN:
        return {
            ...state,
            ...action.payload,
        };
    default:
        return state;
    }
}
