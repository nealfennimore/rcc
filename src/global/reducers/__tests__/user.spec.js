import { setUsername, setToken } from 'global/actions/user';
import user from '../user';

describe( 'User Reducer', () => {
    it( 'should have default state', () => {
        expect( user( undefined, {} ) ).toMatchSnapshot();
    } );
    it( 'should set username', () => {
        expect( user( undefined, setUsername( 'username' ) ) ).toMatchSnapshot();
    } );
    it( 'should set token', () => {
        expect( user( undefined, setToken( 'token' ) ) ).toMatchSnapshot();
    } );
} );
