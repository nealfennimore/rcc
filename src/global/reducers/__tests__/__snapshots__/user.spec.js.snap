// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Reducer should have default state 1`] = `
Object {
  "token": "",
  "username": "",
}
`;

exports[`User Reducer should set token 1`] = `
Object {
  "token": "token",
  "username": "",
}
`;

exports[`User Reducer should set username 1`] = `
Object {
  "token": "",
  "username": "username",
}
`;
